### ISSUE TITLE
Shapes-style identicon

### ISSUE THREAD

**Issue by fgaz** [fgaz/identicon-styles#1]
Something like the gravatar's ones

**Comment by gdka** [comment]
Hi, I am interested in contribute with this issue for the hacktoberfest:). I was also thinking in adding geometric figures.

**Comment by fgaz** [comment]
Hi and thank you for the interest :)

I'll elaborate a bit more what I had in mind.

The end result should basically look like this:

[![](https://www.gravatar.com/avatar/0000000000000000000000000000f000?s=512&r=any&default=identicon&forcedefault=1)](https://www.gravatar.com/avatar/0000000000000000000000000000f000?s=512&r=any&default=identicon&forcedefault=1)

(change any character from the hash in the URL to generate a new one)

From that image it's quite clear that it's a 4x4 grid of squares. There are only 3 different types of square in an image:
* one for the corners, rotated 90° each time
* one for the central squares, again rotated
* one for the remaining slots, again rotated

The squares themselves do not appear to be constructed randomly. Instead, it seems like they are _picked_ randomly from the list of 45 possible patterns [in this article](https://web.archive.org/web/20190403132644/http://scott.sherrillmix.com/blog/blogger/wp_identicon/)

So basically the algorithm should be something like this:
* pick 3 patterns from the list of possible patterns (~3 bytes)
* pick a starting rotation for each of the 3 patterns (2*3 bits = ~1 byte)
* pick a random color (3 bytes)
* draw the squares in the appropriate places/rotations

You can base the structure of the package off the `squares` one.

I don't think this style requires any parameters, so no need for [fancy type level stuff](https://github.com/fgaz/identicon-styles/blob/0f2aba3b3c5f762865c4d7b5f3a0099add68bb91/squares/Graphics/Identicon/Styles/Squares.hs#L44-L64) this time :)

Finally, please ping me if you need any help!

**Comment by gdka** [comment]
Thank you! I will start to work on this!
